packages/spektate/src/IDeployment.ts

```diff
@@ -429,3 +443,29 @@ export const fetchAuthor = (
     }
   });
 };
+
+export const fetchPR = (
+  repository: IGitHub | IAzureDevOpsRepo,
+  prId: string,
+  accessToken?: string
+): Promise<IPullRequest | undefined> => {
+  return new Promise((resolve, reject) => {
+    if ("username" in repository) {
+      githubGetPR(repository, prId, accessToken)
+        .then((pr: IPullRequest | undefined) => {
+          resolve(pr);
+        })
+        .catch(error => {
+          reject(error);
+        });
+    } else if ("org" in repository) {
+      adoGetPR(repository, prId, accessToken)
+        .then((pr: IPullRequest | undefined) => {
+          resolve(pr);
+        })
+        .catch(error => {
+          reject(error);
+        });
+    }
```

Comment:
what if username and org are not in repository? Promise.resolve and Promise.reject functions will not be called and we have a broken Promise :-). I think we can have a 
```
} else {
  reject(Error(".<some error messages>"));
}
```

Reply:
They should always be in repository, but will add